src/main/scala/wdlTools/types/WdlTypes.scala

```diff
@@ -34,7 +42,20 @@ object WdlTypes {
   case class T_Optional(t: T) extends T
 
   // a user defined structure
-  case class T_Struct(name: String, members: Map[String, T]) extends T
+  sealed trait T_Struct extends T {
+    val name: String
+    val members: Map[String, T]
+  }
+
+  case class T_StructDef(name: String, members: Map[String, T]) extends T_Struct
```

Comment:
Converting T_Struct to a strait with two sub-structures is too much. Without it we aren't able to handle invalid struct definitions, but the cost is readability. I prefer the code to be readable.